src/app/downloads/downloads-table.component.ts

```diff
@@ -0,0 +1,136 @@
+        }
+        this.filter = newFilter;
+        this.buildEpisodeMetrics();
+      }
+    });
+
+    this.allEpisodesSub = this.store.select(selectEpisodes).subscribe((allEpisodes: EpisodeModel[]) => {
+      const allPodcastEpisodes = filterAllPodcastEpisodes(this.filter, allEpisodes);
+      if (allPodcastEpisodes) {
+        this.episodes = allPodcastEpisodes;
+        this.buildEpisodeMetrics();
+      }
+    });
+
+    this.episodeMetricsStoreSub = this.store.select(selectEpisodeMetrics).subscribe((episodeMetrics: EpisodeMetricsModel[]) => {
+      this.episodeMetrics = filterEpisodeMetrics(this.filter, episodeMetrics, 'downloads');
+      this.buildEpisodeMetrics();
+    });
+  }
+
+  buildEpisodeMetrics() {
+    if (this.episodes && this.episodeMetrics && this.episodeMetrics.length) {
+      this.episodeTableData = this.episodeMetrics
+        .filter((epMetric) => !!this.episodes.find(ep => ep.id === epMetric.id))
+        .map((epMetric) => {
+          const downloads = metricsData(this.filter, epMetric, 'downloads');
+          const episode = this.episodes.find(ep => ep.id === epMetric.id);
+          if (episode && epMetric && downloads) {
+            return {
+              title: episode.title,
+              publishedAt: episode.publishedAt,
+              releaseDate: dayMonthDate(episode.publishedAt),
+              id: epMetric.id,
+              downloads: mapMetricsToTimeseriesData(downloads),
+              totalForPeriod: getTotal(downloads)
+            };
+          }
+        })
+        .sort((a, b) => {
+          return moment(new Date(b.publishedAt)).valueOf() - moment(new Date(a.publishedAt)).valueOf();
+        });
+      this.setTableDates();
+    }
+  }
+
+  ngOnDestroy() {
+    if (this.filterStoreSub) { this.filterStoreSub.unsubscribe(); }
+    if (this.allEpisodesSub) { this.allEpisodesSub.unsubscribe(); }
+    if (this.episodeMetricsStoreSub) { this.episodeMetricsStoreSub.unsubscribe(); }
+  }
+
+  isPodcastChanged(state: FilterModel): boolean {
+    return state.podcast && (!this.filter || !this.filter.podcast ||  this.filter.podcast.seriesId !== state.podcast.seriesId);
+  }
+
+  isEpisodesChanged(state: FilterModel): boolean {
+    return state.episodes && (!this.filter || !this.filter.episodes ||
+      state.episodes.every(episode => this.filter.episodes.map(e => e.id).indexOf(episode.id) !== -1) ||
+      this.filter.episodes.every(episode => state.episodes.map(e => e.id).indexOf(episode.id) !== -1));
+  }
+
+  setTableDates() {
+    if (this.episodeTableData && this.episodeTableData[0] && this.episodeTableData[0].downloads) {
+      this.dateRange = this.episodeTableData[0].downloads.map(d => this.dateFormat(new Date(d.date)));
+    }
+  }
+
+  dateFormat(date: Date): string {
+    if (this.filter && this.filter.interval) {
+      switch (this.filter.interval.key) {
+        case INTERVAL_DAILY.key:
+          return dayMonthDate(date);
```

Comment:
Users may want that day of the week date format on the table too (a producer requested it), but we can wait for user feedback on that.